File: LeetCode/101-200/199. 二叉树的右视图.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    List<Integer> list = new ArrayList<>();
    public List<Integer> rightSideView(TreeNode root) {
        dfs(root,0);
        return list;
    }

    public void dfs(TreeNode root,int depth) {
        if (root == null) return;
        if (depth == list.size()) list.add(root.val);
        depth++;
        dfs(root.right,depth);
        dfs(root.left,depth);
    }
}

File: LeetCode/101-200/199. 二叉树的右视图.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
 func rightSideView(root *TreeNode) []int {
    res := make([]int, 0)
    var dfs func (*TreeNode,int)
    dfs = func (node *TreeNode,deep int) {
        if node == nil {
            return
        }
        if len(res) == deep {
            res = append(res, node.Val)
        }
        deep++
        dfs(node.Right,deep)
        dfs(node.Left,deep)
    }
    dfs(root,0)
    return res
}